USWDS-Site: Changelog prototype

amyleadem opened the pull request:
## Related issue
Related to https://github.com/uswds/uswds-site/issues/1280

## Preview link
Preview links:
- Accessibility page
- Accordion page

## Problem statement
_Summarize the problem this PR solves in a clear and concise statement._
<!--
A successful problem statement conveys:
1. The desired state,
2. The actual state, and
3. Consequences of remaining in the current state (who does this affect and to what degree?)
-->

## Solution

### What to include
- Any change that requires user action
    - Markup changes, breaking changes, 
- Any change to visual appearance
- Any change to behavior
- Additions 

### What to exclude
- Code enhancements that do not require user action

## Testing and review
_Share recommended methods for reviewing this change._
<!--
1. Describe the tests that you ran to verify your changes,
2. Provide instructions to reproduce these tests, and
3. Clarify the type of feedback you are looking for at this phase.
-->

---

Before opening this PR, make sure you’ve done whichever of these applies to you:

- [ ] Confirm that this code follows the [18F Front End Coding Style Guide](https://pages.18f.gov/frontend/) and [Accessibility Guide](https://pages.18f.gov/accessibility/checklist/).
- [ ] Run `git pull origin [base branch]` to pull in the most recent updates from your base and check for merge conflicts. (Often, the base branch is `develop`).
- [ ] Run `npm run prettier:scss` to format code.
- [ ] Run `npm test` and confirm that all tests pass.
- [ ] Run your code through [HTML_CodeSniffer](http://squizlabs.github.io/HTML_CodeSniffer/) and make sure it’s error free.

mejiaj commented on _includes/component-changelog.html:
```diff
@@ -0,0 +1,9 @@
+<ul class="usa-content-list">
+    <li>Aug. 5, 2022 - <a href="https://github.com/uswds/uswds-site/pull/1737">Added type="button" to the component code and guidance.</a></li>
+    <li>July 20, 2022 - <a href="https://github.com/uswds/uswds-site/pull/1718">Update documentation to use the correct multi-select attribute.</a> The attribute was updated from aria-multiselectable="true"` to data-allow-multiple.</li>
+</ul>
```
This is a good case where we'd _want_ to consume a markdown file to display on site.

Might not be necessary to link to the CHANGELOG.md link, unless we plan to limit the number of items that we show. 

It'd be nice to see the changes in the sidenav too.

1. Do we plan to limit the number of items?
2. Do we want to include version numbers?
3. Can you show an example of a breaking change?

mejiaj reviewed (commented):
I like the structure so far. It shows the importance of good release notes. Added some questions on component changelogs.

jaclinec commented:
This looks great! My feedback for this proto is similar to the feedback I left for #207. I'd add a line for rationale/research to back up our recommendation where applicable.

amyleadem commented:
Some next steps identified during the team discussion:
- [ ] Improve summary copy
    - Current copy limits list scope - can we more accurately communicate will be included in the list of updates?
- [ ] Figure out how to approach the data structure for changelog items
    - Correct data structure could give us flexibility for different ways to collate later
- [ ] Investigate alternate visual presentation --> pursue possible table layout
- [ ] Include related version number on each item
- [ ] Update guidance to include items that are functional improvements/enhancements
    - We can communicate when components improve, even if there is no related user action

amyleadem commented:
@thisisdano @mejiaj @bonnieAcameron @jaclinec 
### Content structure/Visual updates
I have made some updates to the content and visual presentation of the on-page changelogs.

1. Updated the on-page prototype to display as a table rather than a list. 
    - In these prototypes, different page types display different table content. Specifically, guidance pages will not have the "Version" and "Breaking" columns that the component pages will have. 
1. Updated the summary a bit to widen the scope of what will be included in the on-page changelog. 
    - Note: This prototype assumes that there will NOT be a 1:1 match for content in the CHANGELOG.md file. If we decide instead to match the CHANGELOG.md, we can probably safely remove the summary and link to the changelog file from this section.

There are also some potentially out-of-scope content ideas that are *kinda sorta related* to changelogs that I wanted to throw out there. I am mostly interested in assessing if these are items we want to include in the Changelog effort:
1. Added a place to put the current version number in the Package section 
1. Added a contribute section at the end to act as a call to action for users to engage/update. This was inspired by @bonnieAcameron flagging the Edit button on [digital.gov](https://digital.gov/about/). 

Curious to hear your opinions - 

- Is the table the best format for this information? 
- Do we like the column content?
- Are all of the elements there that we need? 
- Does it cover all possible use cases?

### Data
I also did a first pass at a possible yaml-based data structure. This was designed as part of the MVP to give us flexibility while we work through the transition to component packages. It also gives us a way to create changelog items for guidance pages. 

The hope is that we can figure out a path that does not require dual maintenance for component changelogs, but I wanted to nail down the UX for each format before we dug deeper into that. (If you want to prioritize data structure first instead, please let me know).

Thoughts on this approach?

### Demos
I have added a couple of active demos and some screenshots for reference. Interested in your thoughts!

[Guidance page demo - Accessibility page](https://federalist-ead78f8d-8948-417c-a957-c21ec5617a57.app.cloud.gov/preview/uswds/uswds-site/al-changelog/documentation/accessibility/#latest-updates)
This version hides the link to the changelog, and removes the version number and breaking changes columns. 
![image](https://user-images.githubusercontent.com/93996430/189176942-b6a04ac1-f919-4ca9-b5dd-fe2f38311be8.png)

[Component page demo - Accordion page](https://federalist-ead78f8d-8948-417c-a957-c21ec5617a57.app.cloud.gov/preview/uswds/uswds-site/al-changelog/components/accordion/#changelog-accordion)
![image](https://user-images.githubusercontent.com/93996430/189759144-aabd95f5-f70a-410b-bba8-049df406271f.png)

jaclinec commented:
I think this looks really good and I like the table format. It's easy to scan and is written in plain language. I like the column categories and can't think of anything to suggest.

amyleadem commented:
Notes: 
- Change date format to YYYY-MM-DD
- Demonstrate the transition from USWDS version to package version
- Consider collating everything into a collected changelog

amyleadem commented:
Closing in favor of https://github.com/uswds/uswds-site/pull/1789, which builds on this PR and now has more content.
